packages/react-native-editor/__device-tests__/gutenberg-editor-block-insertion.test.js

```diff
@@ -60,6 +60,10 @@ describe( 'Gutenberg Editor tests for Block insertion', () => {
 		await paragraphBlockElement.click();
 
 		await editorPage.addNewParagraphBlock();
+
+		// wait for accessibility ids to update
+		await driver.sleep( 1000 );
```

Comment:
I'm not sure what other options might exist in the native testing, but in web code it's typically discouraged to use arbitrary timeouts, due to (a) slowing down test runtime and (b) not guaranteeing to always work ([example](https://github.com/WordPress/gutenberg/pull/22189/files#diff-be08cbef4c1bc3475ab657deec947d1fL321-R323)). In Puppeteer tests, we'd instead actually express this as whatever it is that's being waited for (e.g. [`waitForSelector`](https://github.com/puppeteer/puppeteer/blob/v3.0.4/docs/api.md#pagewaitforselectorselector-options) instead of [`waitFor( timeout )`](https://github.com/puppeteer/puppeteer/blob/v3.0.4/docs/api.md#pagewaitforselectororfunctionortimeout-options-args)). Are there any equivalent options here?

Reply:
We're using xpath to find our elements on native so we could indeed use [`driver.waitForElementByXPath`](https://github.com/admc/wd/blob/master/doc/api.md) to speed up tests on faster devices.

Tbh we have quite many anti-patterns on those native e2e tests that we're hoping to fix. This PR was supposed to only be a simple port from gutenberg-mobile so we were not supposed to touch this code but since we did I guess we could improve the code a little bit there 👍

Reply:
@ceyhun Do we really need that sleep given that you're setting an implicit wait timeout of 10s [here](https://github.com/WordPress/gutenberg/pull/22189/files#diff-be08cbef4c1bc3475ab657deec947d1fR45)?

Reply:
This should be gone now, updated the tests with the ones from gutenberg-mobile/develop. `setImplicitWaitTimeout` was also initially added there and I had increased it.